src/osd/osd_types.cc

```diff
@@ -1882,7 +1885,12 @@ void pg_pool_t::decode(bufferlist::const_iterator& bl)
   }
 
   if (struct_v >= 4) {
-    decode(flags, bl);
+    uint64_t eflags;
+    decode(eflags, bl);
+    if (struct_v < 29) {
+      eflags &= ~(FLAG_FULL_QUOTA_OBJECTS|FLAG_FULL_QUOTA_BYTES);
+    }
+    flags = eflags;
```

Comment:
i don't think we need this part.. if we're decoding an old struct then it won't have the flag set, right?

Reply:
Not if a v14.2.1 monitor encoded FLAG_FULL_QUOTA_OBJECTS for a v14.2.0 monitor, then disappeared or restarted or something?

Reply:
Oh, hmm, right.  I think this will lead to more instead of less problems because simply decoding and reencoding a nautilus-era map may change it.  I think instead we're best with what you started with and no filtering of flags at all!  Sorry for the noise

Reply:
Hrm, I take it back (again).

I still think we should drop this.  For the most part 14.2.1 and 14.2.0 doesn't matter because 14.2.0 doesn't have a decode check and ignores this flag.  So if your case happens, where 14.2.1 encodes the new flag, a 14.2.0 mon will silently keep it and pass it along. that's true both for a 14.2.0 peon, or a 14.2.1 leader that takes over the leader role.   Same for 14.2.0 osds.. they will take the new flag and ignore it.

Reply:
So we *do* want to drop this commit?
I mean, I think that will work. The failure case is the leader going from a 14.2.1 to a 14.2.0 monitor and incorrectly maintaining the QUOTA_FULL flags; but the next time a 14.2.1 monitor is leader it will:
* notice that the flags don't match
* fall into the "update" case after noticing a state mismatch between current and prior QUOTA_FULL states
* erroneously set the FULL flag in that update case
* ...not notice that anything is wrong since the smaller quota settings will match, and never fix this.

Shoot, I thought it would converge on the correct settings. I'll have to update it a little bit.